src/main/java/core/basesyntax/service/operation/BalanceOperationHandler.java

```diff
@@ -0,0 +1,14 @@
+package core.basesyntax.service.operation;
+
+import core.basesyntax.service.FruitTransaction;
+import java.util.Map;
+
+public class BalanceOperationHandler implements OperationHandler {
+    @Override
+    public void handle(FruitTransaction transaction, Map<String, Integer> storage) {
+        String fruit = transaction.getFruit();
+        int amount = transaction.getAmount();
+
+        storage.put(fruit, storage.getOrDefault(fruit, amount));
```

Comment:
There seems to be a bug in this line. According to the `getOrDefault()` method documentation, it returns the value to which the specified key is mapped, or defaultValue if this map contains no mapping for the key. In your case, if the map does not contain the key `fruit`, it will return `amount` instead of 0. So, the balance operation should set the new amount regardless of the old one. You should just put the new amount in the storage without calling `getOrDefault()`.